rotkehlchen/externalapis/etherscan_like.py

```diff
@@ -0,0 +1,328 @@
+    return [x[0] for x in sorted(hashes, key=operator.itemgetter(1))]
+
+
+class EtherscanLikeApi(ABC):
+    """Base class for any APIs similar to etherscan."""
+
+    def __init__(
+            self,
+            database: 'DBHandler',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        self.db = database
+        self.msg_aggregator = msg_aggregator
+
+    @overload
+    @abstractmethod
+    def _query(
+            self,
+            chain_id: SUPPORTED_CHAIN_IDS,
+            module: Literal['account'],
+            action: Literal['txlistinternal', 'txlist', 'tokentx'],
+            options: dict[str, Any] | None = None,
+            timeout: tuple[int, int] | None = None,
+    ) -> list[dict[str, Any]]:
+        ...
+
+    @overload
+    @abstractmethod
+    def _query(
+            self,
+            chain_id: SUPPORTED_CHAIN_IDS,
+            module: str,
+            action: str,
+            options: dict[str, Any] | None = None,
+            timeout: tuple[int, int] | None = None,
+    ) -> list[dict[str, Any]] | str | int | dict[str, Any] | None:
+        ...
+
+    @abstractmethod
+    def _query(
+            self,
+            chain_id: SUPPORTED_CHAIN_IDS,
+            module: str,
+            action: str,
+            options: dict[str, Any] | None = None,
+            timeout: tuple[int, int] | None = None,
+    ) -> list[dict[str, Any]] | str | int | dict[str, Any] | None:
+        ...
+
+    @abstractmethod
+    def get_blocknumber_by_time(
+            self,
+            chain_id: SUPPORTED_CHAIN_IDS,
+            ts: Timestamp,
+            closest: Literal['before', 'after'] = 'before',
+    ) -> int:
+        ...
+
+    def _process_timestamp_or_blockrange(
+            self,
+            chain_id: SUPPORTED_CHAIN_IDS,
+            period: TimestampOrBlockRange,
+            options: dict[str, Any],
+    ) -> dict[str, Any]:
+        """Process TimestampOrBlockRange and populate call options"""
+        if period.range_type == 'blocks':
+            from_block = period.from_value
+            to_block = period.to_value
+        else:  # timestamps
+            from_block = self.get_blocknumber_by_time(
+                chain_id=chain_id,
+                ts=period.from_value,  # type: ignore
+                closest='before',
+            )
+            to_block = self.get_blocknumber_by_time(
+                chain_id=chain_id,
+                ts=period.to_value,  # type: ignore
+                closest='before',
+            )
+
+        options['startblock'] = str(from_block)
+        options['endblock'] = str(to_block)
+        return options
+
+    @staticmethod
+    def _maybe_paginate(
+            result: list[dict[str, Any]],
+            options: dict[str, Any],
+    ) -> dict[str, Any] | None:
+        """Check if the results have hit the pagination limit.
+        If yes adjust the options accordingly. Otherwise signal we are done"""
+        if len(result) != ETHERSCAN_QUERY_LIMIT:
```

Comment:
yeah so as I mentioned above this limit shuld be configurable depending on the indexer. Can't assume same value for all